catalog/rocks/catalog.go

```diff
@@ -0,0 +1,244 @@
+
+	// Dereference translates Ref to the corresponding CommitID
+	Dereference(ctx context.Context, repositoryID RepositoryID, ref Ref) (CommitID, error)
+
+	// GetBranch returns the Branch metadata object for the given BranchID
+	GetBranch(ctx context.Context, repositoryID RepositoryID, branchID BranchID) (*Branch, error)
+
+	// SetBranch points the given BranchID at the given Branch metadata
+	SetBranch(ctx context.Context, repositoryID RepositoryID, branchID BranchID, branch Branch) error
+
+	// DeleteBranch deletes the branch
+	DeleteBranch(ctx context.Context, repositoryID RepositoryID, branchID BranchID) error
+
+	// GetCommit returns the Commit metadata object for the given CommitID
+	GetCommit(ctx context.Context, repositoryID RepositoryID, commitID CommitID) (*Commit, error)
+
+	// AddCommit stores the Commit object, returning its ID
+	AddCommit(ctx context.Context, repositoryID RepositoryID, commit Commit) (CommitID, error)
+
+	// ListBranches lists branches
+	ListBranches(ctx context.Context, repositoryID RepositoryID, from BranchID) (BranchIterator, error)
+
+	// FindMergeBase returns the merge-base for the given CommitIDs
+	// see: https://git-scm.com/docs/git-merge-base
+	// and internally: https://github.com/treeverse/lakeFS/blob/09954804baeb36ada74fa17d8fdc13a38552394e/index/dag/commits.go
+	FindMergeBase(ctx context.Context, repositoryID RepositoryID, commitIDs ...CommitID) (*Commit, error)
+
+	// Log returns an iterator that reads all parents up to the first commit
+	Log(ctx context.Context, repositoryID RepositoryID, from CommitID) (CommitIterator, error)
+}
+
+// CommittedManager reads and applies committed snapshots
+// it is responsible for de-duping them, persisting them and providing basic diff, merge and list capabilities
+type CommittedManager interface {
+	// GetEntry returns the provided path, if exists, from the provided TreeID
+	GetEntry(ctx context.Context, ns StorageNamespace, treeID TreeID, path Path) (*Entry, error)
+
+	// ListEntries takes a given tree and returns an EntryIterator seeked to >= "from" path
+	ListEntries(ctx context.Context, ns StorageNamespace, treeID TreeID, from Path) (EntryIterator, error)
+
+	// Diff receives two trees and a 3rd merge base tree used to resolve the change type
+	// it tracks changes from left to right, returning an iterator of Diff entries
+	Diff(ctx context.Context, ns StorageNamespace, left, right, base TreeID, from Path) (DiffIterator, error)
+
+	// Merge receives two trees and a 3rd merge base tree used to resolve the change type
+	// it applies that changes from left to right, resulting in a new tree that
+	// is expected to be immediately addressable
+	Merge(ctx context.Context, ns StorageNamespace, left, right, base TreeID) (TreeID, error)
+
+	// Apply is the act of taking an existing tree (snapshot) and applying a set of changes to it.
+	// A change is either an entity to write/overwrite, or a tombstone to mark a deletion
```

Comment:
Define what a tombstone will look like for entries.

Reply:
Relevant because we removed `IsTombstone` :-/